src/main/java/pt/tqs/g205/services/EncomendaService.java

```diff
@@ -0,0 +1,190 @@
+
+  @Autowired
+  private PratoService pratoService;
+
+  @Autowired
+  private EstadoEncomendaService estadoEncomendaService;
+  
+  @Autowired
+  private EncomendaRestauranteService encomendaRestauranteService;
+
+  @Autowired
+  private EstadoEncomendaRepository estadoEncomendaRepo;
+
+  @Autowired
+  private EstadoEncomendaHoraRepository eehRepo;
+
+  @Autowired
+  private PratosPorEncomendaRepository ppeRepo;
+
+  /**
+   * Obter a lista de todas as encomendas.
+   * @return lista de encomendas.
+   */
+  public List<Encomenda> getAll() {
+    List<Encomenda> encomendas = encomendaRepo.findAll();
+
+    encomendas.forEach(e -> {
+      Iterator<PratosPorEncomenda> it = e.getPratos().iterator();
+
+      while (it.hasNext()) {
+        Prato p = it.next().getPrato();
+        p.setIngredientes(null);
+        p.setRestaurante(null);
+
+      }
+    });
+
+    return encomendas;
+  }
+
+  /**
+   * Obter as encomendas de um cliente.
+   * @param clienteId id do cliente.
+   * @return lista das encomendas do cliente.
+   * @throws NoSuchElementException se o cliente nao existir.
+   */
+  public List<Encomenda> getByClienteId(Integer clienteId) throws NoSuchElementException {
+    List<Encomenda> encomendas = encomendaRepo.getByClienteId(clienteId);
+
+    if (encomendas == null || encomendas.size() == 0)
+      throw new NoSuchElementException();
+
+    encomendas.forEach(e -> {
+      Iterator<PratosPorEncomenda> it = e.getPratos().iterator();
+
+      while (it.hasNext()) {
+        PratosPorEncomenda ppe = it.next();
+        ppe.setId(null);
+        Prato p = ppe.getPrato();
+        p.setIngredientes(null);
+        p.setRestaurante(null);
+      }
+
+      Iterator<EstadoEncomendaHora> estadosIt = e.getEstados().iterator();
+
+      while (estadosIt.hasNext()) {
+        EstadoEncomendaHora eeh = estadosIt.next();
+        eeh.setId(null);
+      }
+
+      e.setCliente(null);
+    });
+
+    return encomendas;
+  }
+
+  /**
+   * Metodo para fazer uma encomenda.
+   * @param cliente id do cliente.
+   * @param encomenda detalhes da encomenda.
+   * @return encomenda.
+   */
+  public Encomenda fazerEncomenda(Integer cliente, EncomendaModel encomenda) {
+
+    Cliente cli = clienteService.getById(cliente);
+    TipoEntrega entrega = tipoEntregaService.getById(encomenda.getTipoEntrega());
+    EstadoEncomenda estado = estadoEncomendaService.getById(1);
+
+    Encomenda enc = encomendaRepo.save(new Encomenda(null, entrega, cli));
+
+    cli.getEncomendas().add(enc);
+    clienteRepo.save(cli);
+    entrega.getEncomendas().add(enc);
+    tipoEntregaRepo.save(entrega);
+
+    EstadoEncomendaHora eeh =
+        new EstadoEncomendaHora(enc, estado, LocalDate.now(), LocalTime.now());
+    eehRepo.save(eeh);
+    estado.getEncomendas().add(eeh);
+    estadoEncomendaRepo.save(estado);
+    enc.getEstados().add(eeh);
+
+    List<Prato> pratos = new ArrayList<>();
+
+    encomenda.getPratos().keySet().forEach(pratoId -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove useless curly braces around statement ([squid:S1602](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1602))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=20&issues=AWPIIg0QkDisRwWBeUf_&open=AWPIIg0QkDisRwWBeUf_)

[ISSUE_KEY]: AWPIIg0QkDisRwWBeUf_